content/ui-ux.md

```diff
@@ -0,0 +1,514 @@
+
+A styleguide consists of two parts:
+
+1. A list of "entries"---a component coupled with a list of specifications; basically a list of different arguments that the component expects which trigger different behaviour.
+
+2. A special route in the development version of the application that renders one or more components with one or more of the specificiations.
+
+For instance, in Galaxy, we have a component styleguide that renders each pure component either one specification at a time, or with all specifications at once. 
+
+[ss]
+
+Such rendering enables very quick development of visual aspects of the component in all possible states. Typically in a complex application, it can be quite difficult to achieve certain states of components purely by "using" the application. For example, in Galaxy, the component screen enters quite a complex state if two deploys to the same application are happening simutaneously. This is a very hard to state to reach for a extended period of time if you are simply using the application!
+
+[ss]
+
+XXX: could say a lot more about this, but I feel like we might just blog about it / do a whole article on it's own once we've had a chance to release some code.
+
+## User Interface Patterns
+
+There are some patterns for building user interfaces that are useful to keep in mind for a Meteor application.
+
+### Responsive Design
+### Accessibility
+
+XXX: Honestly I'm really not sure there's very much Meteor specific in these two headings. I'm not convinced they belong in the guide
+
+
+### Internationalization
+
+Internationalization (i18n) is the process of generalizing the UI of your app in such a way that it's easy to render all text in a different language. In Meteor, the excellent `tap:i18n` package provides an API for building translations and using them in your templates and elsewhere.
+
+#### Using `tap:i18n`
+
+XXX: I'm going to leave this section for now as it's kind of dependent on modules (it's more complex than maybe it needs to be in Todos right now due to all-packages).
+
+#### Places to translate
+
+It's useful to consider the various places in the system that user-readable strings can come out of and make sure that you are properly using the i18n system to generate those strings in each case.
+
+The obvious place for such strings is HTML -- the content of components that the user sees. Additionally any alerts or other messages that are generated on the client side should be translated also.
+
+Another place to be aware of is messages that are generated by the server that are user visible. An obvious place is emails and any other server generated messages (such as mobile push notifications), but more subtle places are return values and error messages on method calls (which should be sent over the wire in a generic form and translated on the client, see above).
```

Comment:
I'd switch the order of these two - first, places to translate, then how to use the specific API.